Nos/Views/CreateUsernameSheet.swift

```diff
@@ -0,0 +1,407 @@
+                    }
+
+                    Spacer(minLength: 0)
+                }
+                .padding(.horizontal, 40)
+
+                NavigationLink {
+                    ExcellentChoicePage(username: usernameObserver.text, isPresented: $isPresented)
+                } label: {
+                    if isVerifying {
+                        ZStack {
+                            ProgressView()
+                                .frame(height: .zero)
+                                .tint(Color.white)
+                            PlainText(.localizable.next)
+                                .hidden()
+                        }
+                    } else {
+                        PlainText(.localizable.next)
+                    }
+                }
+                .padding(.horizontal, 40)
+                .buttonStyle(BigActionButtonStyle())
+                .disabled(verified != true || isVerifying || invalidInput)
+            }
+        }
+        .sheetPage()
+    }
+
+    private func usernameAlreadyClaimedText() -> some View {
+        PlainText(.localizable.usernameAlreadyClaimed)
+            .font(.clarity(.medium, textStyle: .subheadline))
+            .foregroundStyle(Color.red)
+            .lineSpacing(3)
+    }
+
+    private var invalidInput: Bool {
+        usernameObserver.text.count < 3
+    }
+
+    private var validationFailed: Bool {
+        verified == false
+    }
+
+    private func verify(_ username: String) async {
+        verified = nil
+
+        guard !username.isEmpty else {
+            return
+        }
+
+        isVerifying = true
+
+        defer {
+            isVerifying = false
+        }
+
+        do {
+            verified = try await namesAPI.verify(username: username)
+        } catch {
+            Log.error(error.localizedDescription)
+        }
+    }
+}
+
+fileprivate struct ExcellentChoicePage: View {
+
+    var username: String
+    @Binding var isPresented: Bool
+    @State private var isClaiming = false
+    @State private var claimError: ClaimError?
+    @Dependency(\.currentUser) var currentUser
+    @Dependency(\.namesAPI) var namesAPI
+
+    private var attributedUsername: AttributedString {
+        AttributedString(
+            username,
+            attributes: AttributeContainer([NSAttributedString.Key.foregroundColor: UIColor(Color.primaryTxt)])
+        ) + AttributedString(".nos.social")
+    }
+
+    private var showAlert: Binding<Bool> {
+        Binding {
+            claimError != nil
+        } set: { _ in
+        }
+    }
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: 20) {
+            Spacer(minLength: 40)
+            if isClaiming {
+                ProgressView()
+                    .tint(Color.accentColor)
+                    .scaleEffect(1.5)
+                    .frame(maxWidth: .infinity, maxHeight: .infinity)
+            } else if let error = claimError {
+                SwiftUI.Text(error.localizedDescription)
```

Comment:
I see a few uses of `SwiftUI.Text` and `SwiftUI.TextField`. Is something requiring us to use the `SwiftUI.` prefix on these?

Reply:
`Font.swift` defines global functions named Text and TextField that aim to ensure that we always use the clarity font but what it actually does is to confuse the code when we want to customize it even more to match the design. If you agree, I can create a ticket to delete all those extensions plus the static var that defines fonts when we have a more flexible function for that.

Reply:
That's why you also find PlainText over there

Reply:
I think a new ticket to address this is a good idea.